### PR TITLE
Feature/location api

### PR DESCRIPTION
**Pull request by surajsetty97**
Preliminary location API and credit card schema

### CODE REVIEW

**Review by toep — APPROVED** [review]

File: packages/backend/src/index.js [hunk 1]
```diff
@@ -265,7 +265,58 @@ process.on('unhandledRejection', err => {
...
+    if (!correctLat(parseFloat(lat)) || !correctLong(parseFloat(long))) {
+      return res.status(HttpStatus.BAD_REQUEST).send('Invalid arguments');
+    }
+
+    // query only active drivers
+    const drivers = await connection.getRepository(Driver).find({ active: 1 });
+    let closestDriver = {};
+    let leastTime = 10000000;
+    await forEach(drivers, async driver => {
+      const result = await axios.get(
```
Comment (vikr01) [inline packages/backend/src/index.js]: I'd prefer this string be in a separate json file or environment variable, so it doesn't get buried in the code.

File: packages/backend/src/index.js [hunk 2]
```diff
@@ -265,7 +265,58 @@ process.on('unhandledRejection', err => {
...
+
+    if (!correctLat(parseFloat(lat)) || !correctLong(parseFloat(long))) {
+      return res.status(HttpStatus.BAD_REQUEST).send('Invalid arguments');
+    }
+
+    // query only active drivers
+    const drivers = await connection.getRepository(Driver).find({ active: 1 });
+    let closestDriver = {};
+    let leastTime = 10000000;
+    await forEach(drivers, async driver => {
```
Comment (vikr01) [inline packages/backend/src/index.js]: Let's not use `p-iteration`, I'd rather avoid a useless dependency for this. [Async iteration is in proposals right now for javascript](https://github.com/tc39/proposal-async-iteration).

If using an array, do:
```javascript
await Promise.all(drivers.map(driver=>async()=>{...
```

If using an object, same thing but with `Object.values`:
```javascript
await Promise.all(Object.values(drivers).map(driver=>async()=>{...
```

[Alternatively, use an `async` generator](http://2ality.com/2016/10/asynchronous-iteration.html).

File: packages/backend/src/index.js [hunk 3]
```diff
@@ -265,7 +265,58 @@ process.on('unhandledRejection', err => {
...
+        }
+      );
+      const duration = result.data.rows[0].elements[0].duration.value;
+      if (duration < leastTime) {
+        leastTime = duration;
+        closestDriver = driver;
+      }
+    });
+
+    if (Object.keys(closestDriver).keys().length === 0) {
```
Comment (vikr01) [inline packages/backend/src/index.js]: I think you made a mistake here adding `.keys()` at the end.

File: packages/backend/package.json [hunk 4]
```diff
@@ -24,6 +25,8 @@
     "forever": "^0.15.3",
     "http-status-codes": "^1.3.0",
     "mysql": "^2.16.0",
+    "ncp": "^2.0.0",
```
Comment (vikr01) [inline packages/backend/package.json]: What was the reason for adding this? I don't see it used anywhere.

File: packages/backend/.eslintrc [hunk 5]
```diff
@@ -7,6 +7,7 @@
   "parserOptions": {
     "ecmaFeatures": {
       "experimentalObjectRestSpread": true,
+      "experimentalDecorators": true,
```
Comment (vikr01) [inline packages/backend/.eslintrc]: Pick `experimentalDecorators` or `legacyDecorators`, but not both.

**Pull request merged**